File: backend-laravel/app/Models/Supplier.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Supplier extends Model
{
    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'id',
        'business_id',
        'name',
        'phone',
        'email',
        'address',
        'balance',
        'is_active',
    ];

    protected $casts = [
        'balance' => 'decimal:2',
        'is_active' => 'boolean',
    ];

    public function business()
    {
        return $this->belongsTo(Business::class);
    }

    public function products()
    {
        return $this->hasMany(Product::class);
    }

    public function purchaseOrders()
    {
        return $this->hasMany(PurchaseOrder::class);
    }
}

File: mobile-flutter/lib/core/models/supplier.dart
class Supplier {
  final String id;
  final String name;
  final String? phone;
  final String? email;
  final String? address;
  final double balance;
  final bool isActive;

  const Supplier({
    required this.id,
    required this.name,
    this.phone,
    this.email,
    this.address,
    required this.balance,
    required this.isActive,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) => Supplier(
    id: json['id']?.toString() ?? '',
    name: json['name'] ?? '',
    phone: json['phone'] as String?,
    email: json['email'] as String?,
    address: json['address'] as String?,
    balance: (json['balance'] ?? 0).toDouble(),
    isActive: json['isActive'] ?? json['is_active'] ?? true,
  );
}
